src/pull.go

```diff
@@ -51,8 +52,47 @@ func (g *Commands) Pull(byId bool) error {
 	cl, clashes, err := pullLikeResolve(g, byId)
 
 	if len(clashes) >= 1 {
-		warnClashesPersist(g.log, clashes)
-		return ErrClashesDetected
+		if g.opts.FixClashes {
+			g.log.Logln("Some clashes found")
+			if g.opts.canPrompt() {
+				if !promptForChanges("Try fixing them [Y/N] ? ") {
+					return ErrClashesDetected
+				}
+			}
+			sort.Sort(ByPath(clashes))
+
+			lastPath := ""
+			i := 0
+			for _, v := range clashes {
+				if v.Path != lastPath {
+					lastPath = v.Path
+					i = 2
+					continue
+				}
+
+				var newPath string
+				for {
+					ext := filepath.Ext(v.Path)
+					name := strings.TrimSuffix(v.Path, ext)
+					newPath = fmt.Sprintf("%v_%d%v", name, i, ext)
+					i++
+
+					var dupCheck *File
+					dupCheck, err = g.rem.FindByPath(newPath)
+					if err != nil && dupCheck == nil {
```

Comment:
We'll need to check if the err found is `ErrPathNotExists` since different errors can be returned here.

```go
If err != nil && err != ErrPathNotExists {
   return err
}

if dupCheck == nil {
   newPath = filepath.Base(newPath)
   break
}
```